### PR TITLE
Add execute response wrappers and fit db http apis with ccore

### PR DESCRIPTION
**Pull request by 1uvu**

### CODE REVIEW

File: ccore/nebula/errors/errors.go [hunk 1]
```diff
@@ -3,7 +3,8 @@ package errors
 import "errors"
 
 var (
-	ErrUnsupportedVersion = errors.New("unsupported version")
-	ErrUnsupported        = errors.New("unsupported")
-	ErrNoEndpoints        = errors.New("no endpoints")
+	ErrUnsupportedVersion    = errors.New("unsupported version")
+	ErrUnsupported           = errors.New("unsupported")
+	ErrNoEndpoints           = errors.New("no endpoints")
+	ErrVersionEstimateFailed = errors.New("version infer failed")
```
Comment (veezhang) [inline ccore/nebula/errors/errors.go:9]: Use `ErrUnsupportedVersion`? And why use the different words `Estimate` and `infer` ?

Comment (1uvu) [reply ccore/nebula/errors/errors.go:9]: Ok, I will use infer to replace the estimate

File: ccore/nebula/helper.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	"strconv"
+	"strings"
+)
+
+func VersionHelper(address string, port int, username string, password string) (Version, error) {
+	host := strings.Join([]string{address, strconv.Itoa(port)}, ":")
+	for _, version := range []Version{
+		V2_5,
+		V2_6,
+		V3_0,
```
Comment (veezhang) [inline ccore/nebula/helper.go:14]: Give priority to the higher version?

Comment (1uvu) [reply ccore/nebula/helper.go:14]: yes, ok.

File: controllers/db.go [hunk 3]
```diff
@@ -2,27 +2,37 @@ package controllers
...
 import (
 	"encoding/json"
+	"github.com/astaxie/beego/logs"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/types"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/service/pool"
 
 	"github.com/astaxie/beego"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/service/dao"
 	"github.com/vesoft-inc/nebula-http-gateway/common"
```
Comment (veezhang) [inline controllers/db.go]: import group, std, vesoft then others.

File: controllers/db.go [hunk 4]
```diff
@@ -32,13 +42,28 @@ type ExecuteRequest struct {
...
-	var params Request
+	var (
+		res    Response
+		params Request
+	)
 	json.Unmarshal(this.Ctx.Input.RequestBody, &params)
-	nsid, err := dao.Connect(params.Address, params.Port, params.Username, params.Password)
+
+	if params.Version == "" {
+		version, err := nebula.VersionHelper(params.Address, params.Port, params.Username, params.Password)
```
Comment (veezhang) [inline controllers/db.go]: How about auto infer version into `nebula` pkg?

File: controllers/db.go (hunk 3 above)
Comment (1uvu) [reply controllers/db.go]: ok

File: controllers/db.go (hunk 4 above)
Comment (1uvu) [reply controllers/db.go]: yes, It's better.

File: ccore/service/dao/dao.go [hunk 5]
```diff
@@ -275,107 +273,107 @@ func getMapInfo(valWarp *nebula.ValueWrapper, _verticesParsedList *list, _edgesP
...
 }
 
 func Disconnect(nsid string) {
-	pool.Disconnect(nsid)
+	pool.Close(nsid)
 }
 
-func Execute(nsid string, gql string) (result ExecuteResult, err error) {
-	result = ExecuteResult{
+func Execute(nsid string, gql string) (ExecuteResult, interface{}, error) {
```
Comment (veezhang) [inline ccore/service/dao/dao.go]: Why return a new `interface{}` value? And the `interface{}` is unclear, the caller is confused about this.

File: ccore/service/pool/pool.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
+package pool
```
Comment (veezhang) [inline ccore/service/pool/pool.go]: move service to `ccore/nebula/gataway`?

I think it's temporary, and will add `Manager` to manager `client` in `ccore/nebula` pkg.


```
ccore
    nebula
        gateway
            pool
            dao
```

File: ccore/nebula/helper.go (hunk 2 above)
Comment (veezhang) [reply ccore/nebula/helper.go:14]: You need to verify it after connected.
ref: https://github.com/vesoft-inc/nebula-go/blob/40030d441885c03c34054391e841e8f59443e0af/connection.go#L75

File: ccore/service/dao/dao.go (hunk 5 above)
Comment (1uvu) [reply ccore/service/dao/dao.go]: the returned interface{} is the runtime panic error. I will add code commit later.

File: ccore/service/pool/pool.go (hunk 6 above)
Comment (1uvu) [reply ccore/service/pool/pool.go]: Ok.

File: ccore/nebula/client_graph.go [hunk 7]
```diff
@@ -26,7 +31,19 @@ func NewGraphClient(endpoints []string, username, password string, opts ...Optio
...
+		return err
+	}
+
+	resp, err := c.graph.VerifyClientVersion()
+	if err != nil {
+		return fmt.Errorf("failed to verify client version: %s", err.Error())
+	}
+	if resp != nil && resp.ErrorCode != nerrors.ErrorCode_SUCCEEDED {
+		return fmt.Errorf("incompatible version between client and server: %s", string(resp.ErrorMsg))
+	}
```
Comment (veezhang) [inline ccore/nebula/client_graph.go:45]: return `ErrVersionConflict` if version conflicts, nor return the raw `err`?

Comment (veezhang) [reply ccore/nebula/client_graph.go:45]: And auto check the next version only  if `err` is `ErrVersionConflict`

File: ccore/nebula/errors/errors.go [hunk 8]
```diff
@@ -6,4 +6,5 @@ var (
 	ErrUnsupportedVersion = errors.New("unsupported version")
 	ErrUnsupported        = errors.New("unsupported")
 	ErrNoEndpoints        = errors.New("no endpoints")
+	ErrMethodNotSupported = errors.New("method not supported")
```
Comment (veezhang) [inline ccore/nebula/errors/errors.go:9]: Use `ErrUnsupported`?

File: ccore/nebula/examples/main.go [hunk 9]
```diff
@@ -2,56 +2,159 @@ package main
...
 func main() {
 	for _, version := range []nebula.Version{
-		nebula.V2_0_0,
-		nebula.V2_5_0,
-		nebula.V2_5_1,
-		nebula.V2_6_0,
+		//nebula.Version2_5,
+		//nebula.Version2_6,
+		//nebula.Version3_0,
+		nebula.VersionAuto,
```
Comment (veezhang) [inline ccore/nebula/examples/main.go:17]: Range multi connection info?

File: ccore/nebula/examples/main.go [hunk 10]
```diff
@@ -2,56 +2,159 @@ package main
 
 import (
 	"fmt"
-
 	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/types"
+	"github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/wrapper"
+	"log"
+	"strings"
```
Comment (veezhang) [inline ccore/nebula/examples/main.go:9]: sort import group ? `std`, `vesoft` and then others.

File: ccore/nebula/factory.go [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+		//NewStepBuilder() types.StepBuilder
+		//NewPointBuilder() types.PointBuilder
+		//NewLineStringBuilder() types.LineStringBuilder
+		//NewPolygonBuilder() types.PolygonBuilder
+		//NewCoordinateBuilder() types.CoordinateBuilder
+		//NewPlanDescriptionBuilder() types.PlanDescriptionBuilder
+		//NewPlanNodeDescriptionBuilder() types.PlanNodeDescriptionBuilder
+		//NewPairBuilder() types.PairBuilder
+		//NewProfilingStatsBuilder() types.ProfilingStatsBuilder
+		//NewPlanNodeBranchInfoBuilder() types.PlanNodeBranchInfoBuilder
```
Comment (veezhang) [inline ccore/nebula/factory.go:34]: remove it if no more use.

**Review by veezhang — APPROVED** [review]
LGTM

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**